src/Nordea/Components/Error.elm

```diff
@@ -0,0 +1,222 @@
+import Html.Styled.Attributes as Attributes exposing (href)
+import Nordea.Components.FlatLink as FlatLink
+import Nordea.Components.Text as Text
+import Nordea.Html exposing (showIf)
+import Nordea.Resources.Colors as Colors
+import Svg.Styled as Svg exposing (Svg)
+import Svg.Styled.Attributes as SvgAttrs exposing (d)
+
+
+type alias Locale =
+    { no : String, se : String, dk : String } -> String
+
+
+type alias ErrorConfig =
+    { errorType : ErrorType, locale : Locale }
+
+
+type ErrorType
+    = InternalServerError
+    | PageNotFound
+
+
+type Error msg
+    = Error ErrorConfig
+
+
+internalServerError : Locale -> Error msg
+internalServerError locale =
+    init { errorType = InternalServerError, locale = locale }
+
+
+pageNotFound : Locale -> Error msg
+pageNotFound locale =
+    init { errorType = PageNotFound, locale = locale }
+
+
+init : ErrorConfig -> Error msg
+init config =
+    Error { locale = config.locale, errorType = config.errorType }
+
+
+view : List (Attribute msg) -> List (Html msg) -> Error msg -> Html msg
+view attributes children (Error config) =
+    let
+        toI18NString textToTranslate =
+            config.locale textToTranslate
+
+        errorDescription =
+            if config.errorType == PageNotFound then
+                texts.pageNotFound.description |> toI18NString
+
+            else
+                texts.internalServerError.description |> toI18NString
+
+        errorAction =
+            if config.errorType == InternalServerError then
+                texts.internalServerError.action |> toI18NString
+
+            else
+                ""
+    in
+    Html.div
+        [ Attributes.css
+            [ Css.backgroundColor Colors.white
+            , Css.width (pct 100)
+            , Css.height (vh 100)
+            , Css.padding (rem 1.5)
+            , Css.displayFlex
+            , Css.flexDirection Css.column
+            , Css.alignItems Css.center
+            ]
+        ]
+        [ Html.div [ Attributes.css [ Css.marginBottom (rem 1) ] ]
+            [ Text.headlineTwo
+                |> Text.withHtmlTag Html.h1
+                |> Text.view
+                    [ Attributes.css [ Css.textAlign Css.center ]
+                    ]
+                    [ Html.text (texts.heading |> toI18NString) ]
+            ]
+        , Html.div
+            [ Attributes.css
+                [ Css.marginBottom (rem 2)
+                , Css.maxWidth (rem 30)
+                , Css.textAlign Css.center
+                ]
+            ]
+            [ Text.bodyTextLight
+                |> Text.view []
+                    [ Html.text errorDescription
+                    ]
+            ]
+        , Html.div [] [ errorSvg ]
+        , showIf
+            (config.errorType
+                == InternalServerError
+            )
+            (viewActionForInternalServerError errorAction)
+        , Html.div
+            (Attributes.css [ Css.maxWidth (rem 30) ] :: attributes)
+            children
+        ]
```

Comment:
I think this makes it easier to use and control, as css you may provide to `Error.view []` will then also affect the supplied children.

```suggestion
        ] ++ children
```

Reply:
The width of extra information should not exceed 30 rem. Do you still want the users of the component to control that themselves?

Reply:
👍